chains/solana/contracts/tests/ccip/ccip_router_test.go

```diff
@@ -9506,6 +9506,189 @@ func TestCCIPRouter(t *testing.T) {
 				})
 			})
 
+			t.Run("When executing a report with oversized data, it fails, but it succeeds when buffered", func(t *testing.T) {
+				transmitter := getTransmitter()
+
+				msgAccounts := []solana.PublicKey{config.CcipLogicReceiver, config.ReceiverExternalExecutionConfigPDA, config.ReceiverTargetAccountPDA, solana.SystemProgramID}
+
+				sourceChainSelector := config.EvmChainSelector
+				message, _ := testutils.CreateNextMessage(ctx, solanaGoClient, t, msgAccounts)
+				message.Data = bytes.Repeat([]byte{42}, 1000)
```

Comment:
Suggestion: run the test with a data that actually has different bytes here and there. That way, you'll actually check that the data isn't scrambled later on, making the check slightly more robust. I get that you're already achieving something similar with the rest of the fields of the execution report (as this only represents that data), but still :shrug: